ambari-server/src/main/resources/common-services/HBASE/0.96.0.2.0/configuration/hbase-env.xml

```diff
@@ -240,6 +240,7 @@ export HBASE_MANAGES_ZK=false
 export HBASE_OPTS="$HBASE_OPTS -XX:+UseConcMarkSweepGC -XX:ErrorFile={{log_dir}}/hs_err_pid%p.log -Djava.security.auth.login.config={{client_jaas_config_file}} -Djava.io.tmpdir={{java_io_tmpdir}} {{zk_security_opts}}"
 export HBASE_MASTER_OPTS="$HBASE_MASTER_OPTS -Xmx{{master_heapsize}} -Djava.security.auth.login.config={{master_jaas_config_file}}"
 export HBASE_REGIONSERVER_OPTS="$HBASE_REGIONSERVER_OPTS -Xmn{{regionserver_xmn_size}} -XX:CMSInitiatingOccupancyFraction={{regionserver_cms_initiating_occupancy_fraction}} -XX:+UseCMSInitiatingOccupancyOnly -XX:ReservedCodeCacheSize=256m -Xms{{regionserver_heapsize}} -Xmx{{regionserver_heapsize}} -Djava.security.auth.login.config={{regionserver_jaas_config_file}}"
+export HBASE_SERVER_JAAS_OPTS="-Djava.security.auth.login.config={{client_jaas_config_file}}"
```

Comment:
This is already added to HBASE_OPTS. Is that not picked up by HBASE?

Reply:
hbck picks it from HBASE_SERVER_JAAS_OPTS it seems.

To reproduce:
```
export HBASE_OPTS="$HBASE_OPTS -Djava.security.auth.login.config=/usr/hdp/current/hbase-master/conf/hbase_client_jaas.conf"
export HBASE_MASTER_OPTS="$HBASE_MASTER_OPTS -Djava.security.auth.login.config=/usr/hdp/current/hbase-master/conf/hbase_master_jaas.conf"
export HBASE_REGIONSERVER_OPTS="$HBASE_REGIONSERVER_OPTS -Djava.security.auth.login.config=/usr/hdp/current/hbase-master/conf/hbase_regionserver_jaas.conf"
```
With above content in hbase-env.sh, hbase hbck fails with "org.apache.zookeeper.KeeperException$NoAuthException: KeeperErrorCode = NoAuth for /hbase-secure/table-lock
"

However, it works fine with below content in hbase-env.sh:
```
export HBASE_OPTS="$HBASE_OPTS -Djava.security.auth.login.config=/usr/hdp/current/hbase-master/conf/hbase_client_jaas.conf"
export HBASE_MASTER_OPTS="$HBASE_MASTER_OPTS -Djava.security.auth.login.config=/usr/hdp/current/hbase-master/conf/hbase_master_jaas.conf"
export HBASE_REGIONSERVER_OPTS="$HBASE_REGIONSERVER_OPTS -Djava.security.auth.login.config=/usr/hdp/current/hbase-master/conf/hbase_regionserver_jaas.conf"
export HBASE_SERVER_JAAS_OPTS="-Djava.security.auth.login.config=/usr/hdp/current/hbase-master/conf/hbase_client_jaas.conf"
```

Reply:
@zeroflag This is the command I tried:
```
/usr/hdp/current/hbase-client/bin/hbase hbck
```

Reply:
@zeroflag Basically, HBase overrides Djava.security.auth.login.config of HBASE_OPTS by HBASE_REGIONSERVER_OPTS unless we specify HBASE_SERVER_JAAS_OPTS with same Djava.security.auth.login.config value as HBASE_OPTS.

JIRA: https://issues.apache.org/jira/browse/HBASE-15145
Patch: https://issues.apache.org/jira/secure/attachment/12783490/hbase-15145_v2.patch
Article: https://community.hortonworks.com/articles/107172/hbase-hbck-fails-with-noauth-and-client-can-not-au.html